examples/failing/3132.purs

```diff
@@ -0,0 +1,18 @@
+-- @shouldFailWith UnknownClass
+module Main where
+
+import Prelude
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE, log)
+
+import ClassDefinitions
+
+data A
+data B
+
+instance inst2 :: C2 A
+instance inst3 :: (C2 a) => C3 a B
```

Comment:
It seems like the effect of having a "hidden" nullary class/instance is to prevent polymorphic instances like this. Note that `C2 A` is totally fine, despite `C1` not being in scope, and `C3 A B` is *also* totally fine. But requiring `C2 a => C3 a B` causes the missing name error.

Reply:
Hm, I'm not sure. It doesn't seem right to me that this code would cause an unknown class error from `C1` not being in scope, and it also doesn't seem right to me that hidden nullary classes can prevent you from writing instances like this.

My suggestion is that we say this code should actually compile successfully, and in fact we need to modify the compiler to make sure that hidden classes like `C1` are made available in the relevant places in the compiler to make sure that internal error doesn't get hit.

Reply:
I think the real problem with all this, is that the `ClassDefinitions` module should demand that you export `C1` because it's required by `C2`. That would be the rule we use for all the other cases for this kind of thing.